### DISCUSSION TITLE
Setting Cache-Control Headers in next.config.js

### DISCUSSION THREAD

**Discussion by mmbmf1** [vercel/next.js#17344] [Help] [answered]
```js
const withPlugins = require('next-compose-plugins')
const withBundleAnalyzer = require('@next/bundle-analyzer')({
  enabled: process.env.ANALYZE === 'true',
})

module.exports = withPlugins([
  [withBundleAnalyzer({})],
  {
    async headers() {
      return [
        {
          source: '/:pages*',
          headers: [
            {
              key: 'Cache-Control',
              value: 'private, max-age=180, must-revalidate',
            },
            {
              key: 'X-test-header',
              value: 'this is a test',
            },
          ],
        },
      ]
    },
  },
])

```

This is my `next.config.js` file. Is this the proper way to set `Cache-Control` `headers` inside `next.config.js` for **ALL** of my `/pages`? This does work locally but **NOT** in deployments.

**Comment by fvpDev** [comment]
Ran into the same issue. "withPlugins" requires the next config to be an object, so change `async headers() {` to `headers: async () => {`. That worked for me.

**Comment by leerob** [comment, marked as answer]
It's now possible to customize `cache-control` as of v14.2.10. Please upgrade and give it a try!

https://github.com/vercel/next.js/pull/69802
